packages/ra-ui-materialui/src/field/ArrayField.tsx

```diff
@@ -79,9 +79,15 @@ const ArrayFieldImpl = <
 >(
     props: ArrayFieldProps<RecordType>
 ) => {
-    const { children, resource, perPage, sort, filter } = props;
+    const { children, resource, perPage, sort, filter, storeKey } = props;
     const data = useFieldValue(props) || emptyArray;
-    const listContext = useList({ data, resource, perPage, sort, filter });
+    const listContext = useList({
+        data,
+        resource: storeKey || resource, // Prioritize storeKey if provided
```

Comment:
I'd prefer that you add support for a custom `storeKey` in `useList`. I know that we currently use `resource` only for selection in `useList`, but this may change in the future.